### PR TITLE
Make dependency of apply_prep before apply clear

### PR DESCRIPTION
**Pull request by npwalker**

### CODE REVIEW

File: documentation/applying_manifest_blocks.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Similar to the `puppet apply` command, which applies a standalone Puppet manifes
 
 ## How manifest blocks are applied
 
-When you run a plan that contains a manifest block, the `apply_prep` function installs the packages necessary to run the Bolt `apply` command.
+When you run a plan that contains a manifest block, 99% of the time you want to include the `apply_prep` function in your plan before your manifest block.  `apply_prep` installs the packages necessary to run the Bolt `apply` command and gathers facts by running `facter` making the facts available to manifest blocks.  Without any facts manifest blocks won't compile.
```
Comment (adreyer) [inline documentation/applying_manifest_blocks.md:19]:
```suggestion
When you run a plan that contains a manifest block, 99% of the time you want to include the `apply_prep` function in your plan before your manifest block. `apply_prep` ensures the necessary packages to run the Bolt `apply` are installed and gathers facts by running `facter` making the facts available to manifest blocks.  Without common facts most manifest blocks won't compile.  
```

Comment (adreyer) [reply documentation/applying_manifest_blocks.md:19]: I don't like 99% but I'm not sure whats better.

Comment (npwalker) [reply documentation/applying_manifest_blocks.md:19]: The alternate wording could be more around what the mechanism is.  "when you run a plan that contains a manifest block, the targets the manifest block is running against need to have facts set on them.  The easiest way to ensure facts are set is to run `apply_prep` on those targets before the manifest block.  ....

Comment (hestonhoffman) [reply documentation/applying_manifest_blocks.md:19]: I'm going to push up a change to this quick.

Comment (hestonhoffman) [reply documentation/applying_manifest_blocks.md:19]: I'm going to push up a change to this

**Comment by adreyer** [comment]
@npwalker is this prominent enough for you with heston's change?

**Comment by npwalker** [comment]
I think the rewrite makes it less prominent.  We should discuss the goal.  

The goal here is to make anyone looking at using an apply block know that they should just put an apply_prep in front of it.

**Comment by hestonhoffman** [comment]
I can turn it into a note if you want it to be highlighted on the page.

File: documentation/applying_manifest_blocks.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ Within a plan, you can use Bolt to apply blocks of Puppet code (manifest blocks)
...
 
+Manifest blocks require facts to compile. If your plan includes a manifest block, use the `apply_prep` function in your plan before your manifest block. The `apply_prep` function installs the packages necessary to run the Bolt `apply` command and gathers facts by running [facter](https://puppet.com/docs/facter/latest/), making the facts available to the manifest block. The `apply_prep` function also identifies the targets that do not have Puppet agents and runs the `puppet_agent::install` task (from the [puppet_agent module](https://forge.puppet.com/puppetlabs/puppet_agent)).
```
Comment (hestonhoffman) [inline documentation/applying_manifest_blocks.md:7]:
```suggestion
> **Important:** Manifest blocks require facts to compile. If your plan includes a manifest block, use the `apply_prep` function in your plan before your manifest block. The `apply_prep` function installs the packages necessary to run the Bolt `apply` command and gathers facts by running [facter](https://puppet.com/docs/facter/latest/), making the facts available to the manifest block. The `apply_prep` function also identifies the targets that do not have Puppet agents and runs the `puppet_agent::install` task (from the [puppet_agent module](https://forge.puppet.com/puppetlabs/puppet_agent)).
```

**Comment by hestonhoffman** [comment]
I think this is as clear as it's going to get for now. I'll revisit this when I revise the plans docs

**Pull request merged**